azurelinuxagent/ga/exthandlers.py

```diff
@@ -783,6 +783,19 @@ def __init__(self, ext_handler, protocol):
             self.logger.error(u"Failed to create extension log dir: {0}", e)
 
         log_file = os.path.join(self.get_log_dir(), "CommandExecution.log")
+
+        try:
+            if os.stat(log_file).st_size > execution_log_max_size:
+                with open(log_file, "rb") as existing_log_file_buffer:
+                    existing_log_file_buffer.seek(-1 * execution_log_max_size, 2)
+                    _ = existing_log_file_buffer.readline()
+
+                    fileutil.write_file("".join([log_file, ".tmp"]), existing_log_file_buffer.read(), asbin=True)
```

Comment:
We shouldn't read the whole file to memory. At this point you can probably use shutil.copyfileobj to copy the rest of the file

Reply:
ah, that works. To be clear, we are only reading the max file size of 10MB into memory at this point, but I think a dedicated copy function is obviously better.